### ISSUE TITLE
oci_database_db_system db_version change cannot be ignored

### ISSUE THREAD

**Issue by matthewdw** [terraform-providers/terraform-provider-oci#814]
Terraform v0.12.3
provider.oci v3.31.0

I have built a "oci_database_db_system" resource but "terraform apply" immediately wants to destroy/create due to changed db_version.  I built it with db_version = "18.0.0.0" but the state now shows it as "18.6.0.0.190416".

I have referenced #767 but this workaround "as is" errors with -

```
Error: Attribute name required

  on db.tf line 39, in resource "oci_database_db_system" "testdb":
  39:       ignore_changes  = ["db_home.0.db_version",]

Dot must be followed by attribute name.
```

Using the following DOES work.

```
    lifecycle {
      ignore_changes  = ["db_home",]
      prevent_destroy = true
  }
```
But I'd like to be able to narrow it down to the version specifically.

**Comment by matthewdw** [comment]
The above workaround seems to work v0.11 but not in v0.12 so would appreciate knowing if there is a way to avoid db_version from wanting to destroy the database without having to ignore the entire db_home object.

**Comment by alexng-canuck** [comment]
Hi @matthewdw , thanks for reporting this.

It appears you have already filed a bug with HashiCorp for the `ignore_changes` workaround in v0.12: https://github.com/hashicorp/terraform/issues/21972

The latest response from @jbardin should unblock the `ignore_changes` workaround in v0.12.

Regarding the unexpected diffs: The provider suppresses differences when a patch version is changed by the service (e.g. when you specify "18.0.0.0" and the DB service returns "18.0.0.0.1", those differences should be suppressed)

But in your case, it seems the DB service has returned a version that is a significantly different than the one that you asked for in your configs. From Terraform standpoint, this diff is expected behavior because your config has drifted significantly from the actual state.

We will need to discuss with the service team on the intended behavior of this attribute and how to fix this.

**Comment by alexng-canuck** [comment]
In the meantime, @matthewdw let us know if the `ignore_changes` workaround is sufficient to unblock you.

**Comment by matthewdw** [comment]
Hi @alexng-canuck,

Thanks for the response.  I've updated hashicorp/terraform#21972 to state that the suggested syntax to ignore the db_version change has not worked.  Will await their response unless you can suggest an alternative.  Will also await your feedback from the service team.

**Comment by jbardin** [comment]
Hi @alexng-canuck,

The [`DiffSuppressFunc` for that attribute](https://github.com/terraform-providers/terraform-provider-oci/blob/master//oci/crud_helpers.go#L548:6) only checks if the config is a simple prefix of the stored value. The linked issue claims that the config shows `db_version = "18.0.0.0"`, but the actual value stored is `18.6.0.0.190416` which doesn't contain the prefix. 

The `depends_on` workaround has a known bug that should be taken care of shortly. While using anything but top-level attributes in `depends_on` was never documented to work, we are making progress ensuring that use case is specified and can work reliably. 

Note that if there are other resources that could reference the `db_version` field in the configuration, they are likely to run into errors ("Diffs didn't match after apply", or now "Provider produced an invalid value for ...") since the value they read also read from the config will be changed during apply.

**Comment by alexng-canuck** [comment]
Yeah, the `DiffSuppressFunc` was written with the assumption that the service value can still match the config's `db_version` as long as they have the same prefix (i.e. the db version is still under the requested version)

What we're seeing in this issue is that the service has returned a wholly different version which doesn't contain the prefix and that has broken this assumption. We need to work with the DB service team to understand why they are doing this. The provider has no control over the value returned by the service, though we could choose to ignore it and not set it in state.

In the meantime, @matthewdw you could update your config to point to exact version that is in the state.

**Comment by abhilash-av** [comment]
@matthewdw, this was released in provider version v3.43.0(https://github.com/terraform-providers/terraform-provider-oci/releases/tag/v3.43.0). Closing this issue for now.
